app/services/history_recorder.rb

```diff
@@ -1,9 +1,23 @@
 class HistoryRecorder
-  def initialize(user:)
+  def initialize(user:, history_model: HistoricalEvent)
```

Comment:
Are we ever going to record anything other than a `HistoricalEvent`? IMO *this* is a premature abstraction that adds a layer of indirection for no benefit.

Reply:
Good question: I've added `HistoricalEvent` as a default parameter in the initialiser because it's a *dependency* (`HistoryRecorder` depends on `HistoricalEvent`) and I believe it makes sense to inject dependencies where possible. Two advantages:

1. testing is easier and faster. We can assert that a mock of `HistoricalEvent` receives the expected `::create` messages -- rather than by using the ActiveRecord stack and even using the database. I find this approach (mocking everything except the class under test) helps focus on the messages passed between objects. Rather than looking at the effect the class under test has had on a collaborating object.
2. If we do need to change or elaborate the dependency (unlikely in this case I think) then we know where to find it (at the top in the constructor) -- and importantly the tests for `HistoryRecorder` should be unaffected by the change

Does this sell it to you?

Reply:
Not really? Testing might be easier, but in my experience it also correlates with making it easier to write tests that still pass even when the functionality got broken. 🤷🏻‍♀️ 

For some reason it also seems to make debugging of live production bugs much harder.

Reply:
Yeah, I'm inclined to agree that making API changes to facilitate easier testing makes me a tad nervous. Could we not do something like:

```
expect(HistoricalEvent).to recieve(:create).with ...
```

Reply:
OK, I've moved this dependency down into the class, hopefully that makes this class easier to read.

Reply:
Yeah, this is much better, and doesn't sacrifice performance too 👍